### PR TITLE
[IMPROVED] Bypass meta-layer's API queue for non-existent consumers.

### PR DESCRIPTION
**Pull request by derekcollison**
Processing a missing consumer was a heavier operation on the meta-layer and could lead to API queue buildup.
This processes that condition at point of entry to the system vs at the meta-layer.
If the consumer assignment does exist we process as normal for now.

Also moved jetstream clustered state and system account to atomics to avoid lock contention.
Add in lookup for both stream and consumer assignments at the same time.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -4731,6 +4727,14 @@ func (js *jetStream) consumerAssignment(account, stream, consumer string) *consu
 	return nil
 }
 
+// Return both the stream and consumer assignments.
```
Comment (neilalexander) [inline server/jetstream_cluster.go:4730]: Maybe add to the comment that the JS lock must be held.

File: server/norace_test.go [hunk 2]
```diff
@@ -11252,3 +11252,59 @@ func TestNoRaceJetStreamClusterLargeMetaSnapshotTiming(t *testing.T) {
...
+
+	// Check to make sure we never have any pending on the API queue.
+	go func() {
+		ml := c.leader()
+		for {
+			select {
+			case <-done:
+				return
+			case <-time.After(100 * time.Millisecond):
+				if qlen := ml.jsAPIRoutedReqs.len(); qlen > 0 {
```
Comment (neilalexander) [inline server/norace_test.go:11282]: May not be an issue but might be a bit more robust if it's `ml.jsAPIRoutedReqs.len() + ml.jsAPIRoutedReqs.inProgress()` as then it'll capture if there's a consumer info pulled off the queue and in progress. (The API worker doesn't `recycle()` until the ICB has finished.)

**Review by neilalexander — APPROVED** [review]
Otherwise LGTM!

File: server/norace_test.go (hunk 2 above)
Comment (derekcollison) [reply server/norace_test.go:11282]: ok will do. Noticed this though which felty weird.

`./norace_test.go:11282:13: invalid operation: ml.jsAPIRoutedReqs.len() + ml.jsAPIRoutedReqs.inProgress() (mismatched types int and int64)`

**Pull request merged**